src/components/ShufflingCardGrid.tsx

```diff
@@ -24,31 +15,46 @@ class ShufflingCardGrid extends Component<any,any> {
       centered: false,
       cardIndex: _.fromPairs(
         this.props.cards.map((c: any, index: any) => [c.key, index])
-      )
+      ),
+      activeCards: this.props.cards.filter( (c: any) => c.mark >= this.props.stage - 1)
```

Comment:
Seemed appropriate as state, since active means "should be rendered on screen" (inactive cards were not selected in a previous stage). Could of course be recomputed in render?